### ISSUE TITLE
YAML null key not supported

### ISSUE THREAD

**Issue by rubensa** [stoplightio/spectral#2276]
**Describe the bug**
I'm not allowed to use the YAML **null** key.

**To Reproduce**
When I try to apply [Configuration key conflicts](https://quarkus.io/guides/config-yaml#configuration-key-conflicts) like:
```yaml
quarkus:
  http:
    cors:
      ~: true
      origins: "*"
```
I get the following problem report:
`Mapping key must be a string scalar rather than null`

**Expected behavior**
The null key should be allowed and no problem should be reported.

**Comment by P0lip** [comment]
Well, that's somewhat expected.
We treat YAML in a similar way to JSON.
If you want, you can remove that warning by setting `parserOptions.incompatibleValues` to `off` in your ruleset.
```js
parserOptions: {
  incompatibleValues: 'off',
}
```

**Comment by P0lip** [comment]
LMK if that did the trick!
Feel free to follow up in case you still run into the issue
